packages/emotion/README.md

```diff
@@ -92,180 +101,242 @@ return (
 </EmotionThemeProvider>
 )}
 ```
+
 You can combine class specific overrides with component specific ones
-  ```jsx
+
+```jsx
 import SomeOtherComponent from  './SomeOtherComponent'
 import { EmotionThemeProvider } from  "@instructure/emotion"
 import { canvas, canvasHighContrast } from  "@instructure/ui-themes"
 
 const  SomeComponentUnderApp = () => {
 const themeOverride = {
-	themes:{
-		canvas:{
-			components:{
-				Avatar:{
-					color:"yellow"
-					}
-				}
-				colors:{
-					textAlert:"pink"
-				}
+themes:{
+canvas:{
+	components:{
+		Avatar:{
+			color:"yellow"
 			}
 		}
+		colors:{
+			textAlert:"pink"
+		}
+	}
+}
 }
 
 return (
 <EmotionThemeProvider theme={themeOverride}>
-	<SomeOtherComponent/>
+<SomeOtherComponent/>
 </EmotionThemeProvider>
 )}
 ```
 
 ## Temporary component migration guide
 
 ### Goal
-Migrate components from themeable-based  to emotion-based styling
+
+Migrate components from themeable-based to emotion-based styling
 
 ### Migration steps
 
-#### 1.
-You need to create a ```styles.js``` file next to the ```theme.js``` and ```styles.css```. You will use the content of the two and merge into one style generator file.
-You should write and export a ```function``` named ```generateStyle```
+#### 1. Refactor the `theme.js`.
+
+It will receive the theme and themeOverride objects, contains the variables mapped for the component, and the theme-specific styles.
+
+```js
+/**
+ * Generates the theme object for the component from the theme and provided additional information
+ * @param  {Object} theme The actual theme object.
+ * @param  {Object} themeOverride User provided overrides of the default theme mapping.
+ * @return {Object} The final theme object with the overrides and component variables
+ */
+const generateComponentTheme = (theme, themeOverride = {}) => {
+  // the props of theme you want to use
+  const { colors, borders, typography, key: themeName } = theme
+
+  // if any styling should depend on the theme itself,
+  // this object should specify it
+  const themeSpecificStyle = {
+    'canvas-high-contrast': {
+      primaryColor: 'red'
+    },
+    canvas: {
+      primaryColor: 'blue'
+    }
+  }
+
+  // maps the theme variables to component specific style variables
+  const componentVariables = {
+    color: colors?.textBrand,
+    background: colors?.backgroundLightest,
+    borderWidthSmall: borders?.widthSmall,
+    borderWidthMedium: borders?.widthMedium,
+    borderColor: colors?.borderMedium,
+    fontFamily: typography?.fontFamily,
+    fontWeight: typography?.fontWeightBold
+  }
+
+  // return with the final theme object of the component
+  // with the added overrides
+  return {
+    ...componentVariables,
+    ...themeSpecificStyle[themeName],
+    ...themeOverride
+  }
+}
+export default generateComponentTheme
+```
```

Comment:
IT would be nice to see here the original `theme.js` too

Reply:
This "temporary migration guide" section is only made for the 4 of us, since we will be doing all the migrating. :) But I'll add the original theme.js, if you think it will make it clearer for us.